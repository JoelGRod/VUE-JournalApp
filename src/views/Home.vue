<template>
  <div class="home">
    <img alt="Vue logo" src="../assets/logo.png">

    <h1>Hello World</h1>
    <hr>

    <button class="btn btn-primary">Primary</button>
    <button class="btn btn-secondary">Secondary</button>
    <button class="btn btn-success">Success</button>
    <button @click="goToDaybook()" 
            class="btn btn-success"
            test-id="daybook-home-button">Go To Daybook</button>

  </div>
</template>

<script>

export default {
  name: 'Home',
  methods: {
    goToDaybook() {
      this.$router.push({ name: 'Daybook-No-Entry' })
    }
  },
  components: { }
}
</script>
