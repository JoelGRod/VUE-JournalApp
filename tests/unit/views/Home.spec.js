import { shallowMount } from "@vue/test-utils";
import Home from "@/views/Home";

describe('Testing Home.vue', () => {

    let wrapper;

    beforeEach(() => {
        wrapper = shallowMount( Home );
    });

    test('should match with the snapshot', () => {
        //Arrange
        // Act
        // Assert
        expect( wrapper.html() ).toMatchSnapshot();
    });

    test('Do click on Go-To-Daybook button must redirect to no-entry path', () => {
        // arrange
        const mockRouter = {
            push: jest.fn()
        };
        const wrapper = shallowMount( Home, {
            global: {
                mocks: {
                    $router: mockRouter
                }
            }
        });
        // Act
        wrapper.find('[test-id="daybook-home-button"]')
            .trigger('click');
        // Assert
        expect( mockRouter.push ).toHaveBeenCalled();
        expect( mockRouter.push ).toHaveBeenCalledWith({ name: 'Daybook-No-Entry' });
    });

})