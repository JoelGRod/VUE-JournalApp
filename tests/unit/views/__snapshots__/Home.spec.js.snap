// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Home.vue should match with the snapshot 1`] = `
<div class="home"><img alt="Vue logo" src="">
  <h1>Hello World</h1>
  <hr><button class="btn btn-primary">Primary</button><button class="btn btn-secondary">Secondary</button><button class="btn btn-success">Success</button><button class="btn btn-success" test-id="daybook-home-button">Go To Daybook</button>
</div>
`;
